Tallies/TallyClerks/collisionClerk_class.f90

```diff
@@ -225,6 +223,25 @@ subroutine reportInColl(self, p, xsData, mem, virtual)
     ! Return if invalid bin index
     if (binIdx == 0) return
 
+    ! Return if collision is virtual but virtual collision handling is off
+    if (self % virtual) then
+      ! Retrieve tracking cross section from cache
+      flux = p % w / xsData % getTrackingXS(p, p % matIdx(), TRACKING_XS)
+
+    else
+      ! Get material pointer
+      mat => neutronMaterial_CptrCast(xsData % getMaterial(p % matIdx()))
+      if (.not.associated(mat)) then
+        call fatalError(Here,'Unrecognised type of material was retrived from nuclearDatabase')
+      end if
+
+      ! Obtain xss
+      call mat % getMacroXSs(xss, p)
+
+      flux = p % w / xss % total
```

Comment:
Not really, because in TMS getTotalMatXS gets you the material temperature majorant, while getMacroXSs retrieves the material macroscopic xss at the relative energy. I should probably comment the code better to emphasize this.

Reply:
I think this point should be elaborated in the doc comment of the `getTotalMatXS` and `getMacroXSs`. It is a bit difficult to understand what cross-sections are obtained bu which call... however I am not sure how to make it cleaner at this moment.

Reply:
Perhaps there is no way around introducing some kind of `getTrackingMatXs`? This double meaning between the `getMacroXS` and `getTotalMatXS` is going to cause confusion and bug down the line I fear. 

All of this obviously gets a bit confusing since the cross-sections have now become random variables. However, this confusion is not specific to TMS since it appears in URR as well. Basically we need to somehow separate the 'sampled' XSs at a particular state of the flight from a deterministic supremum on their values  (`TrackingXS`?)

Reply:
Yeah I agree that it should be made more clear. I tried to keep the code as simple as possible but clearly it's not an easy task and I might have overcomplicated interpretation. 
The fact that colling `getTotalMatXS` is not safe is not great. I'll have a look.

Reply:
I don't think separating deterministic vs random xss is a good way to do it. When using URR there is no point in asking for the deterministic xss. Same with TMS, one either wants the temperature majorant or the relative energy xss. What I am thinking for now is: 

```
updateTrackingMatXS:
if TMS -> get Temperature majorant
else   -> get Total

updateTotal: 
if TMS -> get relative energy xss (when doing the relative energy sampling one might as well get the whole package) and return total
else   -> get Total as always

updateMacroXSS:
if TMS -> get relative energy xss
else   -> get Macro as always 
```
So as far as TMS is concerned, updateTrackingMatXS gets the deterministic majorant and updateTotal and updateMacro give the relative energy xss. When TMS is not used, there is no difference between updateTracking and updateTotal.